### PR TITLE
Fixed #23790: Added warning to AppConfig.label docs reference.

### PR DESCRIPTION
**Pull request by nanorepublica**
# Trac ticket number

ticket-23790

# Branch description
Adds a docs warning to `AppConfig.label` about renaming the label in the middle of a project and how that creates breaking changes to migrations.

Screenshot 2024-06-28 at 15 16 26

# Checklist
- [x] This PR targets the `main` branch. 
- [x] The commit message is written in past tense, mentions the ticket number, and ends with a period.
- [x] I have checked the "Has patch" ticket flag in the Trac system.
- [x] I have added or updated relevant tests.
- [x] I have added or updated relevant docs, including release notes if applicable.
- [x] I have attached screenshots in both light and dark modes for any UI changes.

### CODE REVIEW

**Comment by aaugustin** [comment]
Thank you for taking care of this. Better 10 years later than never...

Based on the content of the ticket, I'd say:

> ... is used in the name of database tables and in migration files.

This gives users a better list of what they need to change if they want to do it anyway.

(Also I'm wondering if a comma is missing there: "or**,** in the case..." — I'm not a native speaker though.)

**Comment by aaugustin** [comment]
Other than that - 👍

I relinquished my commit access so someone else will have to push the merge button 🙂

**Comment by nanorepublica** [comment]
Changes have been made! @aaugustin would you be so kind as to mark the ticket as 'Ready for Checkin'?

**Review by nessita — APPROVED** [review]
Thank you @nanorepublica for this work! :star2: 

I have pushed a small change and ensured that the new docs are wrapped at 79 cols. Will merge when CI is green.

**Pull request merged**
